packages/SwingSet/src/controller.js

```diff
@@ -37,6 +43,51 @@ function makeConsole(tag) {
   return harden(cons);
 }
 
+function makeStartXSnap(bundles, { env }) {
+  const xsnapOpts = {
+    os: osType(),
+    spawn,
+    stdout: 'inherit',
+    stderr: 'inherit',
+    debug: !!env.XSNAP_DEBUG,
+  };
+
+  const pool = defaultSnapstorePath({ env });
+  fs.mkdirSync(pool, { recursive: true });
+
+  const snapStore = makeSnapstore(pool, {
+    tmpName,
+    existsSync: fs.existsSync,
+    createReadStream: fs.createReadStream,
+    createWriteStream: fs.createWriteStream,
+    rename: fs.promises.rename,
+    unlink: fs.promises.unlink,
+    resolve: path.resolve,
+  });
+
+  let supervisorHash = '';
+  return async function startXSnap(name, handleCommand) {
+    if (supervisorHash) {
+      return snapStore.load(supervisorHash, async snapshot => {
+        const xs = xsnap({ snapshot, name, handleCommand, ...xsnapOpts });
+        await xs.evaluate('null'); // ensure that spawn is done
+        return xs;
+      });
+    }
+    const worker = xsnap({ handleCommand, name, ...xsnapOpts });
+
+    for await (const [it, superCode] of Object.entries(bundles)) {
```

Comment:
Given how important it is that the bundles are installed in the correct order, I'd prefer to have `bundles` be an array instead of an object. I know `Object.entries` iterates in insertion order, but that's decidedly not obvious, and explicit is better than implicit.